### PR TITLE
feat(anta.tests): MLAG tests update

### PR DESCRIPTION
**Pull request by carl-baillargeon**
Hello,

Please review and test these updated/new MLAG tests.

Thank you

### CODE REVIEW

File: anta/tests/mlag.py [hunk 1]
```diff
@@ -1,93 +1,233 @@
...
+        """
+        Run VerifyMlagConfigSanity validation
+        """
 
         command_output = cast(Dict[str, Dict[str, Any]], self.instance_commands[0].output)
 
-        if "mlagActive" not in command_output.keys():
-            self.result.is_error("Incorrect JSON response - mlagActive state not found")
-        elif command_output["mlagActive"] is False:
+        if command_output["mlagActive"] is False:
```
Comment (gmuloc) [inline anta/tests/mlag.py:112]: Why is this gone:

```
if "mlagActive" not in command_output.keys():
            self.result.is_error("Incorrect JSON response - mlagActive state not found")
```

I guess it was certainly there for some corner case we may not be able to repro today but we shoudl keep it

File: anta/tests/mlag.py [hunk 2]
```diff
@@ -1,93 +1,233 @@
...
             self.result.is_skipped("MLAG is disabled")
-        elif len(command_output["globalConfiguration"]) > 0 or len(command_output["interfaceConfiguration"]) > 0:
-            self.result.is_failure()
-            if len(command_output["globalConfiguration"]) > 0:
-                self.result.is_failure("MLAG config-sanity returned Global inconsistancies: " f"{command_output['globalConfiguration']}")
-            if len(command_output["interfaceConfiguration"]) > 0:
-                self.result.is_failure("MLAG config-sanity returned Interface inconsistancies: " f"{command_output['interfaceConfiguration']}")
+            return
```
Comment (gmuloc) [inline anta/tests/mlag.py:114]: why return here and using `elif` afterwards ?

File: anta/tests/mlag.py [hunk 3]
```diff
@@ -1,93 +1,233 @@
...
+
+        if len(command_output["globalConfiguration"]) == 0 and len(command_output["interfaceConfiguration"]) == 0:
+            self.result.is_success()
+
+        elif len(command_output["globalConfiguration"]) > 0:
+            self.result.is_failure(f"MLAG config-sanity returned Global inconsistancies: {command_output['globalConfiguration']}")
+
+        elif len(command_output["interfaceConfiguration"]) > 0:
+            self.result.is_failure(f"MLAG config-sanity returned Interface inconsistancies: {command_output['interfaceConfiguration']}")
```
Comment (gmuloc) [inline anta/tests/mlag.py:123]: This is not the same as was in the test originallly

Here you cannot catch both interfce and global inconsistencies so it should be changed back to what it was

Comment (gmuloc) [inline anta/tests/mlag.py:123]: we should probably add sommething if we reach the last elif otherwise (or is it actually possible ? I think not so maybe we should rewrite it in another way)

File: anta/tests/mlag.py [hunk 4]
```diff
@@ -1,93 +1,233 @@
...
+        if detection_delay is None or errdisabled is None or recovery_delay is None or recovery_delay_non_mlag is None:
+            self.result.is_skipped(
+                f"{self.__class__.name} did not run because detection_delay, errdisabled, recovery_delay or recovery_delay_non_mlag were not supplied"
+            )
+            return
+
+        if errdisabled:
+            action = "errdisableAllInterfaces"
         else:
+            action = "none"
```
Comment (gmuloc) [inline anta/tests/mlag.py:209]:
```suggestion
        errdisabled_action = "errdisableAllInterfaces" if errdisabled else "none"
```

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: anta/tests/mlag.py [hunk 5]
```diff
@@ -1,93 +1,234 @@
...
-                self.result.is_failure("MLAG config-sanity returned Interface inconsistancies: " f"{command_output['interfaceConfiguration']}")
+            return
+
+        keys_to_verify = ["globalConfiguration", "interfaceConfiguration"]
+        verified_output = {key: get_value(command_output, key) for key in keys_to_verify}
+
+        if not any(verified_output.values()):
+            self.result.is_success()
         else:
+            self.result.is_failure(f"MLAG config-sanity returned inconsistancies: {verified_output}")
```
Comment (gmuloc) [inline anta/tests/mlag.py:127]:
```suggestion
            self.result.is_failure(f"MLAG config-sanity returned inconsistencies: {verified_output}")
```
